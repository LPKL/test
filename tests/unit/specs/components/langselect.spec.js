import { shallowMount } from '@vue/test-utils'
import LangSelect from '@/components/LangSelect'
import mountOption from '~/cores/vueinstance'

describe('语言选择组件', () => {
  let wrapper
  const msg = '中文 English'
  beforeEach(() => {
    wrapper = shallowMount(LangSelect, mountOption({ msg }))
  })
  test('语言选择文字内容', () => {
    expect(wrapper.text()).toMatch(msg)
  })
  test('选择中文后的变量', () => {
    const vm = wrapper.vm
    vm.handleSetLanguage('zh')
    expect(vm.language).toMatch('zh')
    expect(vm.$i18n.locale).toMatch('zh')
    expect(vm.$store.getters.language).toMatch('zh')
  })
  test('语言选择快照', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})

// import Vue from 'vue'
// import LangSelect from '@/components/LangSelect'
// // import App from '@/App.vue'

// describe('App.test.js', () => {
//   let Cmp, vm

//   beforeEach(() => {
//     Cmp = Vue.extend(LangSelect) // Create a copy of the original component
//     vm = new Cmp({
//       data: {
//         // Replace data value with this fake data
//         messages: ['Cat']
//       }
//     }).$mount()// Instances and mounts the component
//   })

//   it('equals messages to ["Cat"]', () => {
//     expect(vm.messages).toEqual(['Cat'])
//   })
// })
