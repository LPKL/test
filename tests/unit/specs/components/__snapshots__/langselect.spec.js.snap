// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`语言选择组件 语言选择快照 1`] = `
<el-dropdown-stub
  class="international"
  hideonclick="true"
  hidetimeout="150"
  msg="中文 English"
  placement="bottom-end"
  showtimeout="250"
  size=""
  tabindex="0"
  trigger="click"
  visiblearrow="true"
>
  <div>
    <svg-icon-stub
      classname="international-icon"
      iconclass="language"
    />
  </div>
   
  <el-dropdown-menu-stub
    appendtobody="true"
    arrowoffset="0"
    boundariespadding="5"
    offset="0"
    placement="bottom"
    popperoptions="[object Object]"
    transformorigin="true"
    visiblearrow="true"
  >
    <el-dropdown-item-stub
      command="zh"
      disabled="true"
    >
      中文
    </el-dropdown-item-stub>
     
    <el-dropdown-item-stub
      command="en"
    >
      English
    </el-dropdown-item-stub>
  </el-dropdown-menu-stub>
</el-dropdown-stub>
`;
